File: test-suite/src/test/java/io/micronaut/servlet/docs/undertow/UndertowServerCustomizer.java
package io.micronaut.servlet.docs.undertow;

// tag::imports[]
import io.micronaut.context.event.BeanCreatedEvent;
import io.micronaut.context.event.BeanCreatedEventListener;
import io.undertow.Undertow;
import jakarta.inject.Singleton;
// end::imports[]
import io.micronaut.context.annotation.Requires;

@Requires(property = "spec.name", value = "UndertowServerCustomizerTest")
// tag::class[]
@Singleton
public class UndertowServerCustomizer implements BeanCreatedEventListener<Undertow.Builder> {
    @Override
    public Undertow.Builder onCreated(BeanCreatedEvent<Undertow.Builder> event) {
        Undertow.Builder undertowBuilder = event.getBean();
        // perform customizations, for example:
        undertowBuilder.setIoThreads(2);
        return undertowBuilder;
    }
}
// end::class[]

File: test-suite-python/src/test/python/micronaut/servlet/docs/undertow/UndertowServerCustomizer.py
# tag::imports[]
from io.undertow import Undertow
from jakarta.inject import Singleton
from micronaut.context.event import BeanCreatedEvent, BeanCreatedEventListener
# end::imports[]
from micronaut.context.annotation import Requires


@Requires(property="spec.name", value="UndertowServerCustomizerTest")
# tag::class[]
@Singleton
class UndertowServerCustomizer(BeanCreatedEventListener[Undertow.Builder]):

    def onCreated(self, event: BeanCreatedEvent[Undertow.Builder]) -> Undertow.Builder:
        undertow_builder = event.getBean()
        # perform customizations, for example:
        undertow_builder.setIoThreads(2)
        return undertow_builder
# end::class[]
